src/ert/_c_wrappers/util/substitution_list.py

```diff
@@ -71,6 +71,18 @@ def indexForKey(self, key):
     def substitute(self, to_substitute: str) -> str:
         return self._alloc_filtered_string(to_substitute)
 
+    def substitute_real_iter(
+        self, to_substitute: str, realization: int, iteration: int
+    ) -> str:
+        if f"<GEO_ID_{realization}_{iteration}>" in self:
+            to_substitute = to_substitute.replace(
+                "<GEO_ID>", f"<GEO_ID_{realization}_{iteration}>"
```

Comment:
Can you maybe elaborate why this is done differently?
Why is one substitution done before the `_alloc_filtered_string`, even if it preserves the string?

Reply:
The GEO_ID is a concept coming from "Everest". The "Everest" reservoir optimization application is using the BatchSimulator from ERT to get its results and it is using GEO_ID to identify one run of the whole ensemble.

So from ERT point of view if we run an ensemble of 2 models for 2 iterations we would have the same GEO_ID for the first batch of runs i.e 

|realization| Iteration| geo_id|
| 0              | 0          |0          |    
| 0              | 1          |0          |    
| 1              | 0          |0          |    

--------------------------------------

A new batch of forward models runs
|realization| Iteration| geo_id|
| 2              | 0          |1          |    
| 2              | 1          |1          |    
| 3              | 0          |1          |    

So in order to achieve the correct substitution I am setting `<GEO_ID_{real}_{iter}>` in the list of available substitutions when Everest starts the batch run process and when we encounter a string containing a <GEO_ID> for a given realization and iteration we first substitute `<GEO_ID>` with`<GEO_ID_{real}_{iter}>` so that the final correct substitution is done.